static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx

```diff
@@ -0,0 +1,261 @@
+    }
+
+    if (fieldValue.kind !== 'function') {
+      return true;
+    }
+
+    // if (isMetricWidget) {
+    //   return true;
+    // }
+
+    const functionName = fieldValue.function[0];
+    const primaryOutput = aggregateFunctionOutputType(
+      functionName as string,
+      option.value.meta.name
+    );
+    if (primaryOutput) {
+      return isLegalYAxisType(primaryOutput);
+    }
+
+    if (option.value.kind === FieldValueKind.FUNCTION) {
+      // Functions are not legal options as an aggregate/function parameter.
+      return false;
+    }
+
+    return isLegalYAxisType(option.value.meta.dataType);
+  };
+
+  const canDelete = fields.length > 1;
+
+  const hideAddYAxisButtons =
+    ([DisplayType.WORLD_MAP, DisplayType.BIG_NUMBER].includes(displayType) &&
+      fields.length === 1) ||
+    ([
+      DisplayType.LINE,
+      DisplayType.AREA,
+      DisplayType.STACKED_AREA,
+      DisplayType.BAR,
+    ].includes(displayType) &&
+      fields.length === 3);
+
+  let fieldContents: React.ReactElement;
+  if (displayType === DisplayType.TOP_N) {
+    const fieldValue = fields[fields.length - 1];
+    fieldContents = (
+      <QueryFieldWrapper key={`${fieldValue}:0`}>
+        <QueryField
+          fieldValue={fieldValue}
+          fieldOptions={generateFieldOptions({organization})}
+          onChange={value => handleTopNChangeField(value)}
+          filterPrimaryOptions={filterPrimaryOptions}
+          filterAggregateParameters={filterAggregateParameters(fieldValue)}
+        />
+      </QueryFieldWrapper>
+    );
+  } else {
+    fieldContents = (
+      <React.Fragment>
+        {fields.map((fieldValue, i) => (
+          <QueryFieldWrapper key={`${fieldValue}:${i}`}>
+            <QueryField
+              fieldValue={fieldValue}
+              fieldOptions={fieldOptions}
+              onChange={value => handleChangeField(value, i)}
+              filterPrimaryOptions={filterPrimaryOptions}
+              filterAggregateParameters={filterAggregateParameters(fieldValue)}
+              otherColumns={fields}
+            />
+            {(canDelete || fieldValue.kind === 'equation') && (
+              <DeleteButton onDelete={event => handleRemove(event, i)} />
+            )}
+          </QueryFieldWrapper>
+        ))}
+        {!hideAddYAxisButtons && (
+          <Actions>
+            <AddButton title={t('Add Overlay')} onAdd={handleAdd} />
+            <AddButton title={t('Add an Equation')} onAdd={handleAddEquation} />
+          </Actions>
+        )}
+      </React.Fragment>
+    );
+  }
+
+  return (
+    <Field
+      data-test-id="y-axis"
+      label={t('Y-Axis')}
+      inline={false}
+      style={{padding: `${space(2)} 0 24px 0`, ...(style ?? {})}}
+      flexibleControlStateSize
+      error={errors?.fields}
+      required
+      stacked
+    >
+      {fieldContents}
+    </Field>
+  );
+}
+
+export default YAxisSelector;
+
+const Actions = styled('div')`
+  & button {
+    margin-right: ${space(1)};
```

Comment:
should we use the component https://github.com/getsentry/sentry/blob/master/static/app/components/buttonBar.tsx instead?

Reply:
I added:
```js
const Actions = styled(ButtonBar)`
  justify-content: flex-start;
  gap: ${space(1)};
`;
```
which is very similar to what you had in your [handleDefaultFields PR](https://github.com/getsentry/sentry/pull/31943/files#diff-8fca7755a0a1ccfcd3c979d0f7e2b7be5e18306fa7bfc34722cf088dcb862f46R244), but I needed to add `gap` to space the buttons apart

Reply:
sure... you can use the gap prop 

https://github.com/getsentry/sentry/blob/9da51572f630609ef3620c6166c43ab4885512d1/static/app/views/dashboardsV2/widgetBuilder/yAxisSelector.tsx#L219